second-edition/src/ch19-01-unsafe-rust.md

```diff
@@ -150,44 +172,46 @@ unsafe {
-at an example of a safe abstraction that uses unsafe code.
-
-### Calling an Unsafe Function or Method
-
-The second type of operation that requires an unsafe block is calls to unsafe
-functions. Unsafe functions and methods look exactly like regular functions and
-methods, but they have an extra `unsafe` out front. That `unsafe` indicates the
-function has requirements we as programmers need to uphold when we call this
-function, because Rust can’t guarantee we’ve met these requirements. By calling
-an unsafe function within an `unsafe` block, we are saying that we’ve read this
-function’s documentations and take responsibility for upholding the function’s
-contracts ourselves.
+<span class="caption">Listing 19-3: Desreferencia indicadores sin procesar 
+    dentro de un `unsafe`bloqueado </span>
+
+Crear un indicador no puede hacer ningún daño; solo cuando se accede al valor que
+apunta a que podrías terminar lidiando con un valor inválido.
+
+Tenga en cuenta también que en el listado 19-1 y 19-3 creamos `* const i32` y` * mut i32`
+indicadores crudos que apuntan a la misma ubicación de memoria, la de `num`. Si
+en cambio, intentamos crear una referencia inmutable y mutable a `num`,
+esto no se habría compilado porque las reglas de propiedad de Rust no permiten una
+referencia mutable al mismo tiempo que cualquier referencia inmutable. 
+Con un indicador crudo podemos crear un indicador mutable y uno inmutable para
+la misma ubicación, y cambiar los datos a través del indicador mutable, potencialmente
+creando una carrera de datos. ¡Ten cuidado!
+
+
+Con todos estos peligros, ¿por qué alguna vez usaríamos indicadores crudos? Un uso principal
+ es cuando se conecta con el código C, como veremos en la siguiente sección
+funciones inseguras. Otro caso es cuando se construyen abstracciones seguras que el
+el prestamista no entiende. Vamos a introducir funciones inseguras y luego mirar
+en un ejemplo de una abstracción segura que usa código inseguro.
+
+
+### Llamar a una función o método inseguro
+
+El segundo tipo de operación que requiere un bloque inseguro es llamado
+funciones inseguras. Las funciones y métodos inseguros se ven exactamente 
+como las funciones regulares y métodos, pero tienen un frente `unsafe` extra.
+Eso `unsafe` indica que la función tiene requisitos que nosotros como 
+programadores necesitamos mantener cuando llamamos esta función, porque 
+Rust no puede garantizar que hemos cumplido con estos requisitos.
+Llamando una función insegura dentro de un bloque `unsafe`, estamos diciendo
+que hemos leído estas documentaciones de la función y asumimos la responsabilidad
+de mantener la función de contratos nosotros mismos.
 
 <!-- Above -- so what is the difference, when and why would we ever use the
 unsafe function? -->
 <!-- Tried to clarify /Carol -->
 
-Here’s an unsafe function named `dangerous` that doesn’t do anything in its
-body:
+Aquí hay una función insegura llamada `dangerous` que no hace nada en su cuerpo:
```

Comment:
may be lot to ask to translate 'dangerous' by 'peligrosa'? Would be easier to read